soap/soap.php

```diff
@@ -261,7 +262,7 @@ public function SoapClient ($wsdl, array $options = null) {}
      * @throws SoapFault A SoapFault exception will be thrown if the wsdl URI cannot be loaded.
      * @since 5.0.1
      */
-    public function __construct ($wsdl, array $options = null) {}
+	public function __construct (#[LanguageLevelTypeAware(["7.0" => "string", "7.1" => "?string"], default: "")] $wsdl, array $options = null) {}
```

Comment:
default type set to `""` doesn't make any sense as in this case type will be taken from phpdoc. So as far as $wsdl can be `string` or `?string` I think it makes sense to set attribute like `#[LanguageLevelTypeAware(["7.1" => "?string"], default: "string")].

Reply:
But before that there was no type in signature. The stub would be wrong for PHP 5. If I understood correctly.

Reply:
That's right. there was no type in signature and currently, also there is no type in signature. Type in the attribute is just a way to clarify a type depending on language version but it is not a replacement for type in signature. So declaring `#[LanguageLevelTypeAware(["7.1" => "?string"], default: "string")]` we will define that by default parameter should be `string` (we can't pass some object here, right?), but since PHP 7.1 it also can be null (or nullable string). Hope this shade some light to this attribute.

Reply:
But then that stub is still wrong for PHP7 So the correct way to do this would be to make duplicate stubs. and in some cases you need at least 3 different stubs to get all possible combinations.

Reply:
>But then that stub is still wrong for PHP7

Why? The stub will provide `string` as parameter type for language level <= 7 and `?string` for language level >=7.1. In case of stubs, it's not required for a type to be declared exactly in the signature. It also can be declared in PhpDoc for example. The difference only is that type declared in PhpDoc is applied for all language levels but a type declared in an attribute is applied for the language level defined in the attribute.

Reply:
Fatal error: Uncaught TypeError: SoapClient::__construct(): Argument #1 ($wsdl) must be of type ?string, int given

It does have types